File: src/event/impl/serverGroup/ServerGroupEvent.php
<?php

namespace pocketcloud\cloud\event\impl\serverGroup;

use pocketcloud\cloud\event\Event;
use pocketcloud\cloud\group\ServerGroup;

abstract class ServerGroupEvent extends Event {

    public function __construct(private readonly ServerGroup $serverGroup) {}

    public function getServerGroup(): ServerGroup {
        return $this->serverGroup;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/group/ServerGroupEvent.java
package de.pocketcloud.cloud.event.impl.group;

import de.pocketcloud.cloud.event.Event;
import de.pocketcloud.cloud.template.group.ServerGroup;
import lombok.Getter;

public abstract class ServerGroupEvent extends Event {

    @Getter
    private final ServerGroup serverGroup;

    public ServerGroupEvent(ServerGroup serverGroup) {
        this.serverGroup = serverGroup;
    }
}